src/Agent.Worker/ContainerOperationProvider.cs

```diff
@@ -606,20 +606,54 @@ private async Task StartContainerAsync(IExecutionContext executionContext, Conta
                         useHostGroupId = true;
                     }
 
-                    if (string.IsNullOrEmpty(containerUserName)) // Create a new user with same UID as on the host
+                    // List of commands
+                    Func<string, string> addGroup;
+                    Func<string, string, string> addGroupWithId;
+                    Func<string, string, string> addUserWithId;
+                    Func<string, string, string, string> addUserWithIdAndGroup;
+                    Func<string, string, string> addUserToGroup;
+
+                    try
+                    {
+                        await DockerExec(executionContext, container.ContainerId, "useradd");
+
+                        addGroup = (groupName) => $"groupadd {groupName}";
+                        addGroupWithId = (groupName, groupId) => $"groupadd -g {groupId} {groupName}";
+                        addUserWithId = (userName, userId) => $"useradd -m -u {userId} {userName}";
+                        addUserWithIdAndGroup = (userName, userId, groupName) => $"useradd -m -g {groupName} -u {userId} {userName}";
+                        addUserToGroup = (userName, groupName) => $"usermod -a -G {groupName} {userName}";
+                    }
+                    catch (Exception ex) when (ex is InvalidOperationException)
```

Comment:
> catch (Exception ex) when (ex is InvalidOperationException) [](http://example.com/codeflow?start=19&length=60)

Just catch(InvalidOperation exception), no need for pattern matching here, unless I'm misisng something